src/main/java/build/buf/protovalidate/internal/evaluator/EvaluatorBuilder.java

```diff
@@ -54,8 +55,7 @@ public class EvaluatorBuilder {
     EXTENSION_REGISTRY.add(ValidateProto.oneof);
   }
 
-  private final Map<Descriptor, Evaluator> evaluatorMap = new HashMap<>();
-  private final ConstraintResolver resolver = new ConstraintResolver();
+  private volatile ImmutableMap<Descriptor, Evaluator> evaluatorCache = ImmutableMap.of();
```

Comment:
I don't have to use immutablemap here, but I thought it was a good indication to make it clear that this cache won't be modified - if we have a cache miss we'll create a new map.